### DISCUSSION TITLE
Issue with Google oauth sign-in when passing an argument to the 'signIn' client function.

### DISCUSSION THREAD

**Discussion by brenocastelo** [nextauthjs/next-auth#5533] [Help] [answered]
### Question 💬

Hello!

I'm trying to implement sign-in with google oauth, but I'm having a problem when passing "google" as argument to the _signIn_ client function.

When clicking to sign-in with google the pages just redirect to itself with the query param _?email=_, and the _session_ is returned as null. If I call the signIn function without argument, it works well.

![Screenshot from 2022-10-08 16-23-32](https://user-images.githubusercontent.com/7110913/194714901-96f8c7e6-4138-4e32-b5fa-fe64cc5980d0.png)

### How to reproduce ☕️

Here is my implementation:
`_app.tsx`
```js
import "../styles/globals.css";
import type { AppProps } from "next/app";
import { SessionProvider } from "next-auth/react";

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <SessionProvider session={pageProps.session}>
	    <Component {...pageProps} />
	  </SessionProvider>
   );
}

export default MyApp;
```

`login`
```js
import Image from "next/image";
import { signIn } from "next-auth/react";

export default function Login() {
  return (	 
    <div className="h-screen">
      {// CODE}
       <button type="submit" onClick={() => signIn("google")}> Google</button>
     </div>
  );
}

```
`[...nextauth].ts`
```ts
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";

export const authOptions = {
	providers: [
		GoogleProvider({
			clientId: process.env.GOOGLE_CLIENT_ID,
			clientSecret: process.env.GOOGLE_CLIENT_SECRET,
		}),
	],
};

export default NextAuth(authOptions);
```

Would be happy if someone could have any tips for handling this issue.

Thanks in advance

**Comment by balazsorban44** [comment, marked as answer]
This is the problematic line:
```ts
  <button type="submit" onClick={() => signIn("google")}> Google</button>
```

When you set `type="submit"` on a `button` element, it will behave like a form post event, which has the default behavior you describe. You can safely remove the `type="submit"` prop here.

**Reply by brenocastelo** [reply]
Hey! I worked, alongside removing changing the `form` tag to a `div`. Although, if you want to keep the `form` tag, just need to set the button's type attribute to 'button'.

Thanks!
